list/letters-filter/index.js

```diff
@@ -0,0 +1,177 @@
+let stylable = require('../../mixin/stylable');
+let type = require('focus').component.types;
+
+//Alphabet letters
+const alphabetLetters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+
+//Lodash utilities functions
+let every = require('lodash/collection/every');
+
+let lettersFilterMixin = {
+    mixins: [stylable],
+    /**
+     * Component display name.
+     */
+    displayName: 'list-letters-filter',
+
+    /**
+     * Letters.
+     * The letters array contains '*' for special chars ans all letters from A to Z.
+     * @type {Array}
+     */
+    letters: alphabetLetters.split(''),
+
+    /** @inheritedDoc */
+    getDefaultProps() {
+        return {
+            /**
+             * Current selected letter.
+             * @type {string}
+             */
+            selectedLetter: '',
+
+            /**
+             * Available letters.
+             * @type {string}
+             */
+            availableLetters: '',
+
+            /**
+             * Symbol used for non alphabetic characters.
+             * @type {string}
+             */
+            extraCharsSymbol: '*',
+
+            /**
+             * Handle letter selection action (must be overridden by calling component).
+             * @param letter
+             */
+            handleLetterSelection: function(letter) { console.warn('selected letter: ', letter); } // Action on selection click
+        };
+    },
+
+    /** @inheritdoc */
+    propTypes: {
+        selectedLetter: type('string'),
+        availableLetters: type('string'),
+        extraCharsSymbol: type('string')
+    },
+
+    /** @inheritdoc */
+    getInitialState(){
+        return{
+            selectedLetter: this.props.selectedLetter
+        }
+    },
+
+    /** @inheritDoc */
+    componentWillMount(){
+        /*
+         Check if there are some extra chars in available letters.
+         If so, render extra chars in the beginning of letters bar
+         */
+        let isLettersOnly = every(this.props.availableLetters.split(''), (letter) => {
+            return this.letters.indexOf(letter) >= 0;
+        });
+
+        if(!isLettersOnly){
+            this.letters.unshift(this.props.extraCharsSymbol);
+        }
+    },
+
+    /** @inheritdoc */
+    componentWillReceiveProps(newProps){
+        this.setState({
+                selectedLetter: newProps.selectedLetter,
+                availableLetters: newProps.availableLetters
+            });
+    },
+
+    /**
+     * Handle letter selection.
+     * @private
+     */
+    _selectionFunction(event) {
+        let letter = event.target.getAttribute('data-letter');
+        if(this._isLetterAvailable(letter)){
+            this.setState({selectedLetter: letter}, () => {this.props.handleLetterSelection(letter)});
+        }
+    },
+
+    /**
+     * Check if a letter is available.
+     * @param {string} letter.
+     * @returns {boolean} letter is available in input data
+     * @private
+     */
+    _isLetterAvailable(letter){
```

Comment:
'concise methods' is available in ES6 (use esnext option) or Mozilla JS extensions (use moz).